packages/flutter/test/cupertino/popup_surface_test.dart

```diff
@@ -0,0 +1,328 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This file is run as part of a reduced test set in CI on Mac and Windows
+// machines.
+@Tags(<String>['reduced-test-set'])
+library;
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+class _FilterTest extends StatelessWidget {
+  const _FilterTest(Widget child, {this.brightness = Brightness.light})
+      : _child = child;
+  final Brightness brightness;
+  final Widget _child;
+
+  @override
+  Widget build(BuildContext context) {
+    final Size size = MediaQuery.sizeOf(context);
+    final double tileHeight = size.height / 4;
+    final double tileWidth = size.width / 8;
+    return CupertinoApp(
+      home: Stack(fit: StackFit.expand, children: <Widget>[
+        // 512 color tiles
+        // 4 alpha levels (0.416, 0.25, 0.5, 0.75)
+        for (int a = 0; a < 4; a++)
+          for (int h = 0; h < 8; h++) // 8 hues
+            for (int s = 0; s < 4; s++) // 4 saturation levels
+              for (int b = 0; b < 4; b++) // 4 brightness levels
+                Positioned(
+                  left: h * tileWidth + b * tileWidth / 4,
+                  top: a * tileHeight + s * tileHeight / 4,
+                  height: tileHeight,
+                  width: tileWidth,
+                  child: ColoredBox(
+                    color: HSVColor.fromAHSV(
+                      0.5 + a / 8,
+                      h * 45,
+                      0.5 + s / 8,
+                      0.5 + b / 8,
+                    ).toColor(),
+                  ),
+                ),
+        Padding(
+          padding: const EdgeInsets.all(32),
+          child: CupertinoTheme(
+            data: CupertinoThemeData(brightness: brightness),
+            child: _child,
+          ),
+        ),
+      ]),
+    );
+  }
+}
+
+void main() {
+  group('Color filter', () {
+    testWidgets('Brightness.light color filter', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const _FilterTest(
+          CupertinoPopupSurface(
+            blurSigma: 0,
+            isSurfacePainted: false,
+            child: SizedBox(),
+          ),
+        ),
+      );
+
+      // Golden displays the color filter effect of the CupertinoPopupSurface
+      // when the ambient brightness is light.
+      await expectLater(
+        find.byType(CupertinoApp),
+        matchesGoldenFile('cupertinoPopupSurface.color-filter.light.png'),
+      );
+    });
+    testWidgets('Brightness.dark color filter', (WidgetTester tester) async {
```

Comment:
nit: Can you add an empty line between tests?

Reply:
Done! Definitely looks better.